templates/rector-github-action-check.yaml

```diff
@@ -0,0 +1,48 @@
+# github action that checks code with Rector
+name: Rector
+
+on:
+    pull_request: null
+
+
+jobs:
+    rector:
+        # Don't run on forks.
+        if: github.repository == '__CURRENT_REPOSITORY__'
+
+        runs-on: ubuntu-latest
+
+        steps:
+            -
+                uses: actions/checkout@v3
+                with:
+                    # needed to trigger workflow after push
+                    token: ${{ secrets.ACCESS_TOKEN }}
+
+            -
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: 8.1
+                    coverage: none
+
+            -   uses: "ramsey/composer-install@v2"
+
+            ## First run Rector - here can't be --dry-run !!! it would stop the job with it and not commit anything in the future
+            -   run: vendor/bin/rector --ansi
+
+            -   run: vendor/bin/ecs check --fix --ansi
+
+            # see https://github.com/EndBug/add-and-commit
+            -
+                # commit only to core contributors who have repository access
+                if: github.event.pull_request.head.repo.full_name == github.repository
+                uses: EndBug/add-and-commit@v7.5.0
```

Comment:
this still using set-output that will no longer work after next May https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/ 

I was creating PR to replace it:

- https://github.com/rectorphp/rector-src/pull/3168

but cause pending workflow (probably need to be re-activated for new workflow), see

https://github.com/rectorphp/rector-src/pull/3168#issuecomment-1341265639

Reply:
@samsonasik Could re-open the PR and change part of code to make Rector apply? Let's see if works now :)

Reply:
@TomasVotruba Ok, re-open, the `private $unusedPrivateProperty;` is on purpose to verify Rector apply

- https://github.com/rectorphp/rector-src/pull/3426